nullaway/src/main/java/com/uber/nullaway/fixserialization/FixSerializationConfig.java

```
@@ -0,0 +1,180 @@
+
+  public final Serializer serializer;
+
+  public FixSerializationConfig() {
+    suggestEnabled = false;
+    suggestEnclosing = false;
+    annotationConfig = new AnnotationConfig();
+    outputDirectory = null;
+    serializer = null;
+  }
+
+  public FixSerializationConfig(
+      boolean suggestEnabled,
+      boolean suggestEnclosing,
+      AnnotationConfig annotationConfig,
+      String outputDirectory) {
+    this.suggestEnabled = suggestEnabled;
+    this.suggestEnclosing = suggestEnclosing;
+    this.outputDirectory = outputDirectory;
+    this.annotationConfig = annotationConfig;
+    serializer = new Serializer(this);
+  }
+
+  /**
+   * Sets all flags based on their values in the configuration file.
+   *
+   * @param configFilePath Path to the serialization config file written in xml.
+   */
+  public FixSerializationConfig(String configFilePath) {
+    Preconditions.checkNotNull(configFilePath);
+    Document document;
+    try {
+      DocumentBuilderFactory factory = DocumentBuilderFactory.newInstance();
+      DocumentBuilder builder = factory.newDocumentBuilder();
+      document = builder.parse(Files.newInputStream(Paths.get(configFilePath)));
+      document.normalize();
+    } catch (Exception e) {
+      throw new RuntimeException(
+          "Error in reading/parsing config at path: " + configFilePath + "\n" + e);
+    }
+    this.outputDirectory =
+        XMLUtil.getValueFromTag(document, "serialization:path", String.class).orElse(null);
+    Preconditions.checkNotNull(
+        this.outputDirectory, "Error in FixSerialization Config: Output path cannot be null");
+    suggestEnabled =
+        XMLUtil.getValueFromAttribute(document, "serialization:suggest", "active", Boolean.class)
+            .orElse(false);
+    suggestEnclosing =
+        XMLUtil.getValueFromAttribute(document, "serialization:suggest", "enclosing", Boolean.class)
+                .orElse(false)
+            && suggestEnabled;
```

Shouldn't we rather be checking that if active is set to false, but enclosing is set to true, then the configuration file is incorrect and we should throw an error?

---

cf3d6d9e36c281e0782f83ec2fb8764722ec1abf